src/v/cluster/members_manager.cc

```diff
@@ -79,63 +79,50 @@ ss::future<> members_manager::start() {
 }
 
 bool members_manager::is_already_member() const {
-    return _members_table.local().get_broker(_self.id()).has_value();
+    return _raft0->config().contains_broker(_self.id());
 }
 
 ss::future<> members_manager::start_config_changes_watcher() {
-    // handle initial configuration
-    auto offset = _raft0->get_latest_configuration_offset();
-    return handle_raft0_cfg_update(_raft0->config(), offset)
-      .then([this, offset] {
-          _last_seen_configuration_offset = offset;
-          if (is_already_member()) {
-              (void)ss::with_gate(_gate, [this] {
-                  return maybe_update_current_node_configuration();
-              }).handle_exception_type([](const ss::gate_closed_exception& e) {
-                  vlog(
-                    clusterlog.debug,
-                    "unable to update node configuration, gate closed - {}",
-                    e);
-              });
-          }
-          return ss::now();
-      })
-      .then([this] {
-          (void)ss::with_gate(_gate, [this] {
-              return ss::do_until(
-                [this] { return _as.local().abort_requested(); },
-                [this]() {
-                    return _raft0
-                      ->wait_for_config_change(
-                        _last_seen_configuration_offset, _as.local())
-                      .then([this](raft::offset_configuration oc) {
-                          return handle_raft0_cfg_update(
-                                   std::move(oc.cfg), oc.offset)
-                            .then([this, offset = oc.offset] {
-                                _last_seen_configuration_offset = offset;
-                            });
-                      })
-                      .handle_exception_type(
-                        [](const ss::abort_requested_exception&) {});
-                });
-          }).handle_exception_type([](const ss::gate_closed_exception&) {});
-      });
+    for (auto& b : _raft0->config().brokers()) {
+        if (b.id() == _self.id()) {
+            continue;
+        }
+        co_await update_broker_client(
+          _self.id(),
+          _connection_cache,
+          b.id(),
+          b.rpc_address(),
+          _rpc_tls_config);
+    }
+
+    if (is_already_member()) {
+        (void)ss::with_gate(_gate, [this] {
```

Comment:
did you mean `try_with_gate` since you're catching the gate closed exception being returned from with_gate?